pkg/alertmanager/amcfg.go

```diff
@@ -258,6 +258,13 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, g
 	}
 	globalAlertmanagerConfig.Global = global
 
+	// This is need to check required fields are set either at global or receiver level at later step.
+	if global != nil {
+		cb.cfg = &alertmanagerConfig{
+			Global: global,
+		}
+	}
```

Comment:
we can probably directly set `cb.cfg`?
```suggestion
	// The following steps need to check that the required fields are set either at global or receiver level.
	cb.cfg = &alertmanagerConfig{
		Global: global,
	}
```

Reply:
without the check it will set the defaults, do we want like that?

```
--- FAIL: TestInitializeFromAlertmanagerConfig (0.00s)
    --- FAIL: TestInitializeFromAlertmanagerConfig/missing_route (0.00s)
        amcfg_test.go:1008: 
                Error Trace:    /Users/jayapriyapai/github.com/slashpai/prometheus-operator/pkg/alertmanager/amcfg_test.go:1008
                Error:          Not equal: 
                                expected: (*alertmanager.alertmanagerConfig)(nil)
                                actual  : &alertmanager.alertmanagerConfig{Global:(*alertmanager.globalConfig)(nil), Route:(*alertmanager.route)(nil), InhibitRules:[]*alertmanager.inhibitRule(nil), Receivers:[]*alertmanager.receiver(nil), MuteTimeIntervals:[]*alertmanager.timeInterval(nil), TimeIntervals:[]*alertmanager.timeInterval(nil), Templates:[]string(nil)}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,10 @@
                                -(*alertmanager.alertmanagerConfig)(<nil>)
                                +(*alertmanager.alertmanagerConfig)({
                                + Global: (*alertmanager.globalConfig)(<nil>),
                                + Route: (*alertmanager.route)(<nil>),
                                + InhibitRules: ([]*alertmanager.inhibitRule) <nil>,
                                + Receivers: ([]*alertmanager.receiver) <nil>,
                                + MuteTimeIntervals: ([]*alertmanager.timeInterval) <nil>,
                                + TimeIntervals: ([]*alertmanager.timeInterval) <nil>,
                                + Templates: ([]string) <nil>
                                +})
```

Reply:
we have a flaw in the test: if the function returns an error as expected, we shouldn't check the config value.

Reply:
Discussed offline regarding the test update and it can be a follow-up PR 

Logged for test update https://github.com/prometheus-operator/prometheus-operator/issues/7193